### DISCUSSION TITLE
Variadic option support

### DISCUSSION THREAD

**Discussion by arkgast** [jmcdo29/nest-commander#348] [Q&A] [answered]
I'm trying to use an option with multiple arguments and the behavior I'm having is not what I could expect according to [Commander variadic option](https://www.npmjs.com/package/commander#variadic-option) section where the parsed option is an array.

The current behavior is that I'm getting just the last option. But all options pass through `parseOptions` method.

```
import { Command, CommandRunner, Option } from "nest-commander";

@Command({ name: 'command', options: { isDefault: true } })
export class BasicCommand implements CommandRunner {
  async run(args: string[], options: any): Promise<void> {
    console.log({ args, options })
  }

  @Option({
    flags: '-o, --opts [options...]',
    description: 'Multiple options',
  })
  parseOptions(options: string) {
    console.log({ options })
    return options
  }
}
```

```
node dist/main -o opt1 opt2 opt3
{ options: 'opt1' }
{ options: 'opt2' }
{ options: 'opt3' }
{ args: [], options: { opts: 'opt3' } }
```

Not sure if I'm doing something wrong or this might be a bug.

**Comment by jmcdo29** [comment]
Oh interesting that that method gets called per passed option and not with all of them at once. I may need to modify some code to take care of that. Could you provide me a small minimum reproduction so I can better track this down?

**Comment by arkgast** [comment, marked as answer]
Sure thing @jmcdo29.

[Here you can find a small demo](https://replit.com/@arkgast/nest-commander-variadic-options#src/basic-command.command.ts) with the behavior I explained above.

**Reply by jmcdo29** [reply]
Awesome. Thank you for this. So I was looking into how commander itself handles a situation like this, and it seems that the callback handler for options is actually like so:
```js
option<T>(flags: string, description: string, fn: (value: string, previous: T) => T, defaultValue?: T): this;
```
Looking at this, we can type our custom decorated methods like so:
```ts
  @Option({
    flags: '-o, --opts [options...]',
    description: 'Multiple options',
  })
  parseOptions(option: string, options: string[] = []) {
    console.log('parseOptions: ', { option })
    options.push(option)
    return options
  }
```
And now we get the desired output
```
parseOptions:  { option: 'opt1' }
parseOptions:  { option: 'opt2' }
parseOptions:  { option: 'opt3' }

running command with next args and opts:
{ args: [], options: { opts: [ 'opt1', 'opt2', 'opt3' ] } }
```

[You can view the forked REPL here](https://replit.com/@jmcdo29/nest-commander-variadic-options#src/basic-command.command.ts)

I'm not sure if there's something we should do at the library level, as wrapping already custom methods in our own methods could get messy, so I think the best approach here would be to document that this is how to handle variadic options. What are your thoughts @arkgast?

**Reply by arkgast** [reply]
Yes, I agree.
I was bit confused about how it could be used, now it's clear. Thanks!

**Reply by jmcdo29** [reply]
Cool, I'll update the docs for this then so it's clear for others. Seems there's no changes necessary to the library
